scale/codec.go

```diff
@@ -70,47 +71,41 @@ func (pe Encoder) PushByte(b byte) error {
 //   zL zL zL 10 / zM zM zM zL / zM zM zM zM / zH zH zH zM					(2**14 ... 2**30 - 1)	(u16, u32)  low LMMH high
 //   nn nn nn 11 [ / zz zz zz zz ]{4 + n}									(2**30 ... 2**536 - 1)	(u32, u64, u128, U256, U512, U520) straight LE-encoded
 // Rust implementation: see impl<'a> Encode for CompactRef<'a, u64>
-func (pe Encoder) EncodeUintCompact(v uint64) error {
-
-	// TODO: handle numbers wider than 64 bits (byte slices?)
-	// Currently, Rust implementation only seems to support u128
-
-	if v < 1<<30 {
-		if v < 1<<6 {
-			err := pe.PushByte(byte(v) << 2)
+func (pe Encoder) EncodeUintCompact(v big.Int) error {
+	if v.Uint64() < 1<<30 {
```

Comment:
I think we need an additional check here for `v.IsUint64()`, since v could be bigger than uint64 or even negative.

Reply:
Could you also add a test case for the proper failing if v is bigger than 2^64 and for v being negative?